go/vt/vttablet/onlineddl/executor.go

```diff
@@ -1260,6 +1279,25 @@ func (e *Executor) initMigrationSQLMode(ctx context.Context, onlineDDL *schema.O
 	return deferFunc, nil
 }
 
+// initConnectionLockWaitTimeout sets the given lock_wait_timeout for the given connection, with a deferred value restoration function
+func (e *Executor) initConnectionLockWaitTimeout(ctx context.Context, conn *connpool.Conn, lockWaitTimeout time.Duration) (deferFunc func(), err error) {
+	deferFunc = func() {}
+
+	if _, err := conn.Exec(ctx, `set @lock_wait_timeout=@@lock_wait_timeout`, 0, false); err != nil {
```

Comment:
Nit, but IMO slightly clearer to always use the return var. i.e. not use `:=` here. Behavior is the same though either way.

Reply:
IMO it would add clarity and intention to specify global here and a comment for those not super famiilar:
```
// Set a connection level SQL variable to the current global value.
set @lock_wait_timeout=@@global.lock_wait_timeout
```

Similar for the defer:
```
set @@session.lock_wait_timeout=@lock_wait_timeout
```

(we are using `@@session` already when we actually set it on line 1290)

Reply:
> Nit, but IMO slightly clearer to always use the return var. i.e. not use := here. Behavior is the same though either way.

My take is the opposite, I always find it more comfortable to be explicit.

Reply:
Now using explicit `@@session.` at all places. In no place do I intend to use `@@global.lock_wait_timeout`. I'm reading the current session value, modify it, restoring it. I don't touch the global setting.